### PR TITLE
Rename ExpensiTextInput

### PR DESCRIPTION
**Pull request by mateusbra**
### Details

Rename component `ExpensiTextInput` to `TextInput`
### Fixed Issues

$ https://github.com/Expensify/App/issues/6760

### Tests

1. Run the app and see that TextInput still working
### QA Steps

1. Run the app and see that TextInput still working
### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Web

#### Mobile Web

Mobile Web

#### Desktop

Desktop

#### iOS

IOS

#### Android

Android

### CODE REVIEW

File: src/styles/styles.js [hunk 1]
```diff
@@ -593,7 +593,7 @@ const styles = {
         height: variables.componentSizeLarge,
     },
 
-    expensiTextInputContainer: {
+    TextInputContainer: {
```
Comment (parasharrajat) [inline src/styles/styles.js:596]: All styles in camelCase

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by mateusbra** [comment]
We had a lint error due to having two style names that are the same:
TextInput and TextInput(expensiTextInput previously).
How you think we should rename it?

**Comment by parasharrajat** [comment]
you can try baseTextInput if that works.

**Comment by puneetlath** [comment]
It looks like we previously did it as `RNText` here: https://github.com/Expensify/App/blob/d28716a694af1bdc9449ec5f5f40e57a7ecc6eb1/src/components/ExpensifyText.js#L5

So maybe that's a good format to follow for when we import these components from React Native?

**Comment by mateusbra** [comment]
> It looks like we previously did it as `RNText` here:
> 
> https://github.com/Expensify/App/blob/d28716a694af1bdc9449ec5f5f40e57a7ecc6eb1/src/components/ExpensifyText.js#L5
> 
> So maybe that's a good format to follow for when we import these components from React Native?


You're saying we should change these imports:
https://github.com/Expensify/App/blob/d28716a694af1bdc9449ec5f5f40e57a7ecc6eb1/src/components/ExpensiTextInput/BaseExpensiTextInput.js#L3-L5
into:
```js
import {
    Animated as AnimatedRN, 
View as ViewRN, 
TouchableWithoutFeedback as TouchableWithoutFeedbackRN, 
Pressable as PressableRN,
} from 'react-native';
```

**Comment by puneetlath** [comment]
Ah perhaps I've misunderstood. Can you help me understand why we need to call the files `BaseExpensiTextInput` instead of just `TextInput`?

**Comment by mateusbra** [comment]
@puneetlath our file structure will be this:
Captura de Tela 2021-12-25 às 18 11 18
we won't call `BaseExpensiTextInput`, we will call `BaseTextInput`

File: src/components/TextInput/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, {forwardRef} from 'react';
+import styles from '../../styles/styles';
+import BaseTextInput from './BaseTextInput';
+import {propTypes, defaultProps} from './baseTextInputPropTypes';
```
Comment (parasharrajat) [inline src/components/TextInput/index.js:4]:
```suggestion
import * as baseTextInputPropTypes from './baseTextInputPropTypes';
```

File: src/components/TextInput/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <BaseTextInput
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+        innerRef={ref}
+        inputStyle={[styles.baseTextInput, styles.textInputDesktop]}
+    />
+));
+
+TextInput.propTypes = propTypes;
+TextInput.defaultProps = defaultProps;
```
Comment (parasharrajat) [inline src/components/TextInput/index.js:16]: This will change too..

File: src/components/TextInput/index.native.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
 import React, {forwardRef} from 'react';
 import styles from '../../styles/styles';
-import BaseExpensiTextInput from './BaseExpensiTextInput';
-import {propTypes, defaultProps} from './baseExpensiTextInputPropTypes';
+import BaseTextInput from './BaseTextInput';
+import {propTypes, defaultProps} from './baseTextInputPropTypes';
```
Comment (parasharrajat) [inline src/components/TextInput/index.native.js:4]: Similar to above.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Small changes...

File: src/components/TextInput/index.native.js (hunk 4 above)
Comment (mateusbra) [reply src/components/TextInput/index.native.js]: ok

File: src/components/TextInput/index.js (hunk 3 above)
Comment (mateusbra) [reply src/components/TextInput/index.js]: ok

File: src/components/TextInput/index.js (hunk 2 above)
Comment (mateusbra) [reply src/components/TextInput/index.js]: ok

**Review by parasharrajat — DISMISSED** [review]
LGTM
cc: @puneetlath

**Review by puneetlath — DISMISSED** [review]
@mateusbra thanks for your patience through the holidays. Everything looks good to me, but you've picked up some merge conflicts. Once you fix those I can quickly re-approve and merge. Thanks!

**Review by parasharrajat — APPROVED** [review]

**Review by puneetlath — APPROVED** [review]

**Pull request merged**